crates/tracing-instrumentation/src/exporter.rs

```diff
@@ -74,94 +205,73 @@ impl<T: SpanExporter + 'static> SpanExporter for UserServiceModifierSpanExporter
             }) = span.attributes.iter().find(|kv| kv.key == RPC_SERVICE_KEY)
             {
                 spans_by_service
-                    .entry(Some(string_value.clone()))
+                    .entry(string_value.clone())
                     .or_default()
                     .push(span);
             } else {
-                spans_by_service.entry(None).or_default().push(span);
+                spans_by_service
+                    .entry(StringValue::from(""))
+                    .or_default()
+                    .push(span);
             }
         }
 
-        let resource = self.resource.load();
-
-        Box::pin(async move {
-            for (service_name, batch) in spans_by_service.into_iter() {
-                {
-                    let mut exporter_guard = match exporter.lock() {
-                        Ok(exporter) => exporter,
-                        Err(_) => {
-                            return Err(TraceError::Other("exporter mutex is poisoned".into()));
-                        }
-                    };
-                    match service_name {
-                        None => exporter_guard.set_resource(&resource),
-                        Some(service_name) => {
-                            exporter_guard.set_resource(&resource.merge(&Resource::new(
-                                std::iter::once(KeyValue::new(SERVICE_NAME, service_name)),
-                            )))
-                        }
-                    }
-                    exporter_guard.export(batch)
-                }
-                .await?;
-            }
-            Ok(())
-        })
-    }
+        for (service_name, batch) in spans_by_service.into_iter() {
+            let mut exporter = self.builder.build().unwrap();
```

Comment:
Building an exporter is now so cheap by making sure that the channel or the http client are shared across all exporters. The shutdown and flush implementation for the otlp span exporter is a noop. so it's safe to create and drop the exporter without leaking resources.

The benchmark shows that the build() takes around 500ns